cnxpublishing/views/admin.py

```diff
@@ -90,3 +95,162 @@ def admin_post_publications(request):
                 })
 
     return {'states': states}
+
+
+@view_config(route_name='admin-add-error-banner', request_method='GET',
+             renderer='cnxpublishing.views:templates/error-banner.html',
+             permission='administer')
+def admin_post_error_banner(request):
+    settings = request.registry.settings
+    db_conn_str = settings[config.CONNECTION_STRING]
+
+    banners = []
+    with psycopg2.connect(db_conn_str) as db_conn:
+        with db_conn.cursor() as cursor:
+            cursor.execute("""\
+                SELECT id, service_state_id, starts, ends, priority, message
+                FROM service_state_messages ORDER BY starts DESC;""")
+            for row in cursor.fetchall():
+                banners.append({
+                    'id': row[0],
+                    'service_state_id': row[1],
+                    'starts': str(row[2]),
+                    'ends': str(row[3]),
+                    'priority': row[4],
+                    'message': row[5],
+                })
+    today = datetime.today()
+    tomorrow = today + timedelta(days=1)
+    return {'start_date': today.strftime("%Y-%m-%d"),
+            'start_time': today.strftime("%H:%M"),
+            'end_date': tomorrow.strftime("%Y-%m-%d"),
+            'end_time': tomorrow.strftime("%H:%M"),
+            'banners': banners}
+
+
+def parse_error_args(request):
+    args = {}
+    args['message'] = request.POST.get('message', 'Error')
+    args['priority'] = request.POST.get('priority', 1)
+    args['type'] = request.POST.get('type', 1)
+
+    today = datetime.today()
+    tomorrow = today + timedelta(days=1)
+    start_date = datetime.strptime(
+        request.POST.get('start_date', today.strftime("%Y-%m-%d")),
+        '%Y-%m-%d').date()
+    start_time = datetime.strptime(
+        request.POST.get('start_time', today.strftime("%H:%M")),
+        '%H:%M').time()
+    end_date = datetime.strptime(
+        request.POST.get('end_date', tomorrow.strftime("%Y-%m-%d")),
+        '%Y-%m-%d').date()
+    end_time = datetime.strptime(
+        request.POST.get('end_time', tomorrow.strftime("%H:%M")),
+        '%H:%M').time()
+    start = datetime.combine(start_date, start_time)
+    end = datetime.combine(end_date, end_time)
+    args.update({'starts': start, 'ends': end})
+    return args
+
+
+@view_config(route_name='admin-add-error-banner-POST', request_method='POST',
+             renderer='templates/error-banner.html',
+             permission='administer')
+def admin_post_error_banner_POST(request):
+
+    settings = request.registry.settings
+    db_conn_str = settings[config.CONNECTION_STRING]
+
+    print(request.POST.keys())
+
+    # If it was a post request to delete
+    if 'delete' in request.POST.keys():
```

Comment:
Part of me wants to not provide a delete - keeping a record of what messages we've had up is not a horrible thing. Someone's going to want to 'clean up' otherwise. If we do delete, I agree a DELETE http request would be best.